src/ops.jl

```diff
@@ -27,7 +27,7 @@ end
 function Base.cat(ts::Tensor{T,N}...; dims = 1) where {T,N}
   ptr = Ref(Ptr{Cvoid}())
   ts_arr = [i.ptr for i in ts]
-  atg_cat(ptr, ts_arr, length(ts_arr), dims - 1)
```

Comment:
@dhairyagandhi96 ,  is it possible that indexing with the `dims` parameter was previously incorrect here?
When concatenating two (10x10x8x2) tensors along the third dimension, we expect a (10x10x16x2) tensor.
Previously, this would return a (10x20x8x2) tensor. (3-1=2 and torch somehow counts the dimensions from right to left, starting at 0)

Reply:
Right, I think these cases should be added to tests